components/AddTransactionModal.tsx

```diff
@@ -0,0 +1,313 @@
+import { Check, ChevronsUpDown } from "lucide-react";
+import {
+    Dialog,
+    DialogClose,
+    DialogContent,
+    DialogDescription,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+} from "@/components/ui/Dialog";
+import {
+    Command,
+    CommandEmpty,
+    CommandGroup,
+    CommandInput,
+    CommandItem,
+} from "@/components/ui/Command";
+import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+} from "@/components/ui/Popover";
+import { format } from "date-fns";
+import { Calendar as CalendarIcon } from "lucide-react";
+import { Calendar } from "@/components/ui/Calendar";
+
+// TODO: refactor to be able to be prefilled with data for updating portfolio name
+export const AddTransactionModal = () => {
+    const tickers: ticker[] = [
+        {
+            value: "apple",
+            label: "Apple",
+            ticker: "AAPL",
+        },
+        {
+            value: "amazon",
+            label: "Amazon",
+            ticker: "AMZN",
+        },
+        {
+            value: "google",
+            label: "Google",
+            ticker: "GOOG",
+        },
+        {
+            value: "microsoft",
+            label: "Microsoft",
+            ticker: "MSFT",
+        },
+        {
+            value: "meta",
+            label: "Meta",
+            ticker: "META",
+        },
+    ];
+    const [logos, setLogos] = useState<{ [ticker: string]: string }>({});
+    const [open, setOpen] = React.useState(false);
+    const [value, setValue] = React.useState("");
+    const [ticker, setTicker] = React.useState("");
+    const [date, setDate] = React.useState<Date>();
+
+    const [stockPrice, setStockPrice] = useState(0);
+    const [quantity, setQuantity] = useState(0);
+
+    type ticker = {
+        value: string;
+        label: string;
+        ticker: string;
+    };
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+    };
+
+    const fetchLogo = async (ticker: string) => {
+        try {
+            const apiKey = process.env.API_KEY;
+            const apiUrl = `https://api.api-ninjas.com/v1/logo?ticker=${ticker}`;
+
+            const response = await axios.get(apiUrl, {
+                headers: {
+                    "X-Api-Key": apiKey,
+                },
+            });
+
+            // Handle the successful response
+            console.log(response.data);
+
+            // Update the logos object with the fetched logo for this ticker
+            setLogos((prevLogos) => ({
+                ...prevLogos,
+                [ticker]: response.data[0]?.image,
+            }));
+
+            // console.log(logos);
+        } catch (error) {
+            console.error("Error:", error);
+        }
+    };
+
+    // const fetchLogo = async (ticker: string) => {
+    //     try {
+    //         // const ticker = "AAPL";
+    //         const apiKey = "7S5YGbLiYc2AyLf78JKrsg==XjsnlMf8Lp9Yw7L9"; // Replace with your actual API key
```

Comment:
exposed api key

Reply:
thanks for the comments! have hidden the key and regenerated it! good catch!